proxy/http/handler.py

```diff
@@ -374,27 +364,30 @@ def _selected_events(self) -> SelectedEventsGeneratorType:
                 readables.append(key.fileobj)
             if mask & selectors.EVENT_WRITE:
                 writables.append(key.fileobj)
-        yield (readables, writables)
-        for fd in events:
-            self.selector.unregister(fd)
+        return (events, readables, writables)
 
-    def _run_once(self) -> bool:
-        with self._selected_events() as (readables, writables):
-            teardown = self.handle_events(readables, writables)
+    async def _run_once(self) -> bool:
+        events, readables, writables = await self._selected_events()
+        try:
+            teardown = await self.handle_events(readables, writables)
             if teardown:
```

Comment:
This should be in the `else:` part of try-except. Besides, it could be simplified to a one-liner `return bool(teardown)`.

Reply:
Good catch, since `handle_events` are supposed to return `teardown` value itself, can simply do `return self.....`